backend/default.nix

```diff
@@ -0,0 +1,44 @@
+{ haskell-nix, runCommand, lib, hlint }:
+let
+  ignoreFilter = path: type:
+    let
+      inherit (lib) removePrefix hasPrefix hasSuffix;
+      relPath = removePrefix (toString ./. + "/") (toString path);
+      baseName = baseNameOf relPath;
+    in
+      !(
+        baseName == ".gitignore" ||
+        hasPrefix "resources" relPath ||
+        hasSuffix ".md" baseName ||
+        hasSuffix ".nix" baseName
+      );
+
+  # Set a constant name for the src closure
+  source = let
+    root = ./.;
+  in builtins.path {
+    name = "edna-backend-src";
+    path = root;
+    filter = name: type:
+      (ignoreFilter name type && lib.cleanSourceFilter name type);
+  };
+
+  project = haskell-nix.stackProject {
+    src = haskell-nix.haskellLib.cleanGit {
+      src = haskell-nix.cleanSourceHaskell { src = ./.; };
+    };
+    modules = [
+      {
+        doHaddock = true;
+        packages.edna = {
+          doHaddock = true;
+          package.ghcOptions = "-Werror";
+        };
+      }
+    ];
+  };
+  library = project.edna.components.library;
+  server = project.edna.components.exes.edna-server;
+  test = project.edna.checks.edna-test;
+  hlint-html = runCommand "hlint.html" {} "${hlint}/bin/hlint ${source} --no-exit-code --report=$out -j";
```

Comment:
1. What is `hlint.html`? Do you understand?
2. Why do we pass `--no-exit-code`? This way it seems very unlikely to notice when it reports anything. If we don't want CI to fail when `hlint` fails, we can mark the corresponding step as optional in the CI config. Just set `soft_fail: true`. I think it's better because this way it will be easier to notice that `hlint` reported something.

Reply:
1. It's the name of the output file generated in the closure for the `hlint-html` derivation. It's an arbitrary name. The important part is the closure itself and its hash, which depends on the inputs defined in `source`.

2. No idea. This looks like what Yegor wrote years ago for Disciplina. I suspect it has just been cargo culted ever since. Feel free to adjust the invocation of `hlint` as you see fit. Just make sure you put something in `$out` somehow.